### PR TITLE
feat: catalog metadata

### PR DESCRIPTION
**Pull request by rephus**
Closes: #10073
### Description:

[Screencast from 24-05-24 10:29:02.webm](https://github.com/lightdash/lightdash/assets/1983672/5759561b-0f6f-43ba-822c-caf4a29abfd3)

[Screencast from 24-05-24 12:08:13.webm](https://github.com/lightdash/lightdash/assets/1983672/7fc65270-2583-4425-b3cd-3ddd47e65402)

### **Previewing tables quickly using my arrow keys** 
[Sketch](https://www.figma.com/proto/rlqMIJ4pm3t8aAsiMd3CZJ/Lightdash-Product-New?node-id=2544-18430&t=eeMfJ6TrYhH4TyAU-0&scaling=min-zoom&page-id=2544%3A18282&starting-point-node-id=2550%3A115315) (use your right and left arrows to navigate it)
- [x] I can open a 'preview mode' for a table, where additional info about this appears in a sidebar (through an info icon) 

**If a table has been selected:**
- [x]  Highlight the selected table in the sidebar - It should be really clear which table the user is previewing
- [x]  Show a sidebar on the right-hand side displaying the Table Name and Description
- [x] Display the description as markdown as the user intended 
- [x]  The sidebar shows me a full list of fields in the selected table

**Navigation:**
- [x]  If a user double-clicks on a table name in the sidebar, it opens up the table in a fresh explore
- [x]  If a user hits Enter on their keyboard whilst previewing the table, it opens up the table in a fresh explore
- [x]  If a user hits the down/up arrow keys, they scan through the list of available tables in the sidebar.
- [x]  Show clear arrow controls at the bottom of the panel (like in the mockup)
- [x]  If a user searches for a term using the search box, they can only 'preview' the tables that are in the search results. (basically, no change to the behaviour here).

**Other**
- This ticket covers testing the UX of navigating through different previews here, so it can be scrappy. I want to test out how wide the panels need to be and just how smooth the experience is. 
- **Out of scope:** Tags and Usage Analytics

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

**Comment by rephus** [comment]
Added permission checks on metadata fields: 

Before: (with age) 
Screenshot from 2024-05-24 11-38-03

AFter: (no age)
Screenshot from 2024-05-24 11-39-18

File: packages/frontend/src/features/catalog/components/CatalogPanel.tsx [hunk 1]
```diff
@@ -69,112 +84,217 @@ export const CatalogPanel: FC<React.PropsWithChildren<Props>> = ({
...
-                    icon={<MantineIcon icon={IconSearch} />}
-                    rightSection={
-                        search ? (
-                            <ActionIcon onClick={() => setSearch('')}>
-                                <MantineIcon icon={IconX} />
-                            </ActionIcon>
-                        ) : null
+    const history = useHistory();
+    // Keyboard navigation
+    useHotkeys(
```
Comment (magnew) [inline packages/frontend/src/features/catalog/components/CatalogPanel.tsx:106]: Nice. I was worried this was table-based, but this should be portable to the tree

File: packages/frontend/src/features/catalog/hooks/useCatalog.ts [hunk 2]
```diff
@@ -24,7 +24,7 @@ export const useCatalog = ({ projectUuid, search, type }: GetCatalogParams) => {
     const setErrorResponse = useQueryError();
 
     return useQuery<ApiCatalogResults, ApiError>({
-        queryKey: ['comments', projectUuid, type, search],
+        queryKey: ['catalog', projectUuid, type, search],
```
Comment (magnew) [inline packages/frontend/src/features/catalog/hooks/useCatalog.ts:27]: Ooops. It feels like this kind of typo is very common 😄

File: packages/frontend/src/features/catalog/components/CatalogPanel.tsx [hunk 3]
```diff
@@ -69,112 +84,217 @@ export const CatalogPanel: FC<React.PropsWithChildren<Props>> = ({
...
+                    onClick={() => {
+                        if (metadata) closeMetadata();
+                        else if (selectedTable === undefined)
+                            selectAndGetMetadata(
+                                ungroupedCatalogItems[0]?.name,
+                            );
+                        else selectAndGetMetadata(selectedTable);
+                    }}
+                >
+                    {metadata ? 'Show metadata' : 'Hide metadata'}
```
Comment (magnew) [inline packages/frontend/src/features/catalog/components/CatalogPanel.tsx:288]: This is backwards. It says show when you can hide. But, this is also obviously very temporary. I can fix this in the next review if you want.

**Review by magnew — APPROVED** [review]
Nice. Love the design. Let's change nothing :D

**Pull request merged**